### DISCUSSION TITLE
Is there a current way of building eCAL from source in Linux other than Ubuntu?

### DISCUSSION THREAD

**Discussion by chengguizi** [eclipse-ecal/ecal#1220] [Q&A]
Since many lower-power embedded compute system on the market does not ship Ubuntu. Some do Debian and some just Yocto. Hence the question: Is there a good way to compile eCAL from source on a generic Linux system?

I found this discussion thread https://github.com/eclipse-ecal/ecal/issues/241 , but it is quite a while back, not sure if anything has changed.

I hope there is no must-have dependencies in eCAL now that only available in Ubuntu :)

**Comment by FlorianReimold** [comment]
Hi @chengguizi 

eCAL is cross plattform and currently compiles on Windows, Linux, macOS, BSD and QNX. It's not at all specific to Ubuntu. However, there are too many Linux distros for us to specifically support all of them, so we just stick with testing on all supported Ubuntu versions. Ubuntu also is the most relevant platform for our internal development, so it makes sense to give it a special place among other Linux distros.

You need a C++14 compliant compiler (clang, gcc or msvc) for the eCAL Core or a C++17 compliant compiler for some apps. On many Linux systems you can get most eCAL build dependencies from some kind of package repository. If your Linux distro doesn't have that, you need to get those through some other way. If you want to compile GUI apps, you obviously need some Desktop environment.

There is no common Linux distro, so we cannot give common advice that would work for all Linux distros.

Here you can take a look at the CMake options that you can use while compiling eCAL in order to turn features  and builtin dependencies on and off:
https://github.com/eclipse-ecal/ecal/blob/b0b7655d9e1ca6d3c532fbfdbd14ddcd2db92e89/.github/workflows/build-ubuntu-22.yml#L59-L96
